analyzers/tests/SonarAnalyzer.UnitTest/SymbolicExecution/Roslyn/RoslynSymbolicExecutionTest.Invocation.cs

```diff
@@ -1260,6 +1255,39 @@ public void Invocation_BoolEquals_BothArgumentsHaveNoConstraint_DoesNotLearn()
         validator.TagValue("Result").Should().HaveOnlyConstraint(ObjectConstraint.NotNull);
     }
 
+    [DataTestMethod]
+    [DataRow("42", "42", true)]
+    [DataRow("0", "42", false)]
+    [DataRow("((int?)42)", "((int?)42)", true)]
+    [DataRow("((int?)42)", "((int?)0)", false)]
+    public void Invocation_NumberEquals_LearnsResult(string left, string right, bool expected)
+    {
+        var code = $"""
+                var result = {left}.Equals({right});
+                Tag("Result", result);
+
+                var resultStaticCall = object.Equals({left}, {right});
+                Tag("ResultStaticCall", resultStaticCall);
+                """;
+        var validator = SETestContext.CreateCS(code).Validator;
+        validator.TagValue("Result").Should().HaveOnlyConstraints(ObjectConstraint.NotNull, BoolConstraint.From(expected));
+        validator.TagValue("ResultStaticCall").Should().HaveOnlyConstraints(ObjectConstraint.NotNull, BoolConstraint.From(expected));
+    }
+
+    [TestMethod]
+    public void Invocation_NumberEquals_DoesNotLearn()
+    {
+        var code = """
+                if (i>1 && i<10)
+                {
+                    var result = object.Equals(i, 0);
+                    Tag("Result", result);
+                }
+                """;
+        var validator = SETestContext.CreateCS(code, "int i").Validator;
+        validator.TagValue("Result").Should().HaveOnlyConstraints(ObjectConstraint.NotNull);
```

Comment:
Add a comment to document the not-implemented behavior.
```suggestion
        validator.TagValue("Result").Should().HaveOnlyConstraints(ObjectConstraint.NotNull);    // Should learn false
```